amethyst_assets/src/simple.rs

```diff
@@ -135,22 +148,30 @@ impl<A> AssetPtr<A>
             self.version = version;
         }
     }
+}
 
-    /// Returns `true` if a clone of this `AssetPtr` exists.
-    pub fn is_shared(&self) -> bool {
-        Arc::strong_count(&self.update) > 1
+/// `Asset` implementation that supports hot reloading
+pub struct SimpleAsset<A>(pub AssetPtr<A, SimpleAsset<A>>);
+impl<A> AsRef<A> for SimpleAsset<A> {
+    fn as_ref(&self) -> &A {
+        self.0.inner()
+    }
+}
+impl<A> AsMut<A> for SimpleAsset<A> {
+    fn as_mut(&mut self) -> &mut A {
+        self.0.inner_mut()
     }
 }
 
 /// A simple implementation of the `Context` trait.
-pub struct SimpleContext<A, D, R, T> {
-    cache: Cache<AssetFuture<AssetPtr<A>>>,
+pub struct SimpleContext<A, D, T> {
+    cache: Cache<AssetFuture<SimpleAsset<A>>>,
     category: Cow<'static, str>,
     load: T,
-    phantom: PhantomData<(D, R)>,
+    phantom: PhantomData<*const D>,
```

Comment:
`R` can be inferred
And `D` isn't owned by `SimpleContext`. If you have `PhantomData<D>` drop checker may think that while `SimpleContext` gets dropped it may access inner `D`
Changing it to `*const D` says that's not the case.
If my explanation confuse you (сhukchi is not a writer), you may read about it [here](https://doc.rust-lang.org/std/marker/struct.PhantomData.html#ownership-and-the-drop-check)